### DISCUSSION TITLE
@sanity/webhook with v3

### DISCUSSION THREAD

**Discussion by safejace** [sanity-io/sanity#3879] [Sanity Studio Support] [answered]
if i am using v3, do i still use @sanity/webhook to handle the secrets?

getting this error:

``` 
ERROR  [worker reload] [worker init] Named export 'SIGNATURE_HEADER_NAME' not found. The requested module 'file:///.../node_modules/@sanity/webhook/lib/index.js' is a CommonJS module, which may not support all module.exports as named exports.
CommonJS modules can always be imported via the default export, for example using:
```

is the cjs a sanity issue, or possible a nuxt 3 issue?

**Comment by kmelve** [comment]
I think this is a Nuxt 3 issue.

**Comment by safejace** [comment, marked as answer]
so I was able to get around it by doing this:

```
import webhook from '@sanity/webhook'
...
const { isValidSignature, SIGNATURE_HEADER_NAME } = webhook
...
```

problem is now that it saying its invalid and i dont know how to debug it:

```
export default defineEventHandler(async (event) => {
  const signature = event.node.req.headers[SIGNATURE_HEADER_NAME]
  const body = await readBody(event)
  console.log('valid?', signature, body,secret)
  if (!isValidSignature(body, signature, secret)) {
    throw createError({
      statusCode: 401,
      statusMessage: 'Invalid signature',
    })
  }

```

everything appears to be set correctly

dunno if @danielroe can chime in here... or if its now back to a sanity issue where the webhook lib doesnt work withe sanity v3 for some reason...?

**Reply by danielroe** [reply]
is the issue here that isValidSignature is looking for the _raw_ body? `readBody` is returning a parsed version.

**Reply by safejace** [reply]
hmmm. ok, is that a nuxt thing? can we get the raw body. will go check the docs now

**Reply by danielroe** [reply]
you can indeed. there is the `readRawBody` util.

**Comment by safejace** [comment]
@kmelve can you help me debug this? i don't think it's a nuxt issue anymore
